site/docs/uk/plugins/runner.md

```diff
@@ -0,0 +1,440 @@
+# Конкурентність (`runner`)
+
+Цей пакет можна використовувати, якщо ви запускаєте бота [використовуючи тривале опитування](../guide/deployment-types.md) та хочете, щоб повідомлення оброблялися паралельно.
+
+> Переконайтеся, що ви розумієте [2-й етап масштабування](../advanced/scaling.md#тривале-опитування), перш ніж використовувати runner.
+
+## Навіщо потрібен runner
+
+Якщо ви розміщуєте бота використовуючи тривале опитування та хочете його масштабувати, не можна обійтися без паралельної обробки оновлень, оскільки послідовна обробка оновлень відбувається надто повільно.
+Унаслідок цього боти стикаються з низкою проблем.
+
+- Чи існують умови гонки?
+- Чи можемо ми все ще чекати (`await`) виконання стеку middleware? Ми мусимо знати це для обробки помилок!
+- Що робити, якщо middleware з якихось причин ніколи не виконається; чи заблокує це бота?
+- Чи можемо ми обробити деякі вибрані оновлення послідовно?
+- Чи можемо ми обмежити навантаження на сервер?
+- Чи можна обробляти оновлення на декількох ядрах?
+
+Як бачите, нам потрібне рішення, яке може вирішити всі перераховані вище проблеми, щоб досягти належного тривалого опитування для бота.
+Це проблема, яка дуже відрізняється від написання middleware або надсилання повідомлень у Telegram.
+Отже, вона не вирішується за допомогою базового пакету grammY.
+Замість цього ви можете використовувати [runner](https://github.com/grammyjs/runner).
+Він також має власну [довідку API](https://deno.land/x/grammy_runner/mod.ts).
+
+## Використання
+
+Ось простий приклад.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+import { Bot } from "grammy";
+import { run } from "@grammyjs/runner";
+
+// Створюємо бота.
+const bot = new Bot("");
+
+// Додаємо звичайний middleware тощо
```

Comment:
```suggestion
// Додаємо звичайний middleware, тощо
```

Reply:
https://onlinecorrector.com.ua/uk/%D0%B7%D0%B0%D0%B9%D0%B2%D0%B0-%D0%BA%D0%BE%D0%BC%D0%B0-%D0%BF%D0%B5%D1%80%D0%B5%D0%B4-%D1%82%D0%BE%D1%89%D0%BE/